Fireworks/Geometry/plugins/TGeoMgrFromDdd.cc

```diff
@@ -46,16 +46,64 @@
 #include "CLHEP/Units/GlobalSystemOfUnits.h"
 #include <cmath>
 
+class TGeoMgrFromDdd : public edm::ESProducer {
+public:
+  TGeoMgrFromDdd(const edm::ParameterSet&);
+  TGeoMgrFromDdd(const TGeoMgrFromDdd&) = delete;
+  const TGeoMgrFromDdd& operator=(const TGeoMgrFromDdd&) = delete;
+
+  using ReturnType = std::unique_ptr<TGeoManager>;
+
+  // ---------- const member functions ---------------------
+
+  // ---------- static member functions --------------------
+
+  // ---------- member functions ---------------------------
+
+  ReturnType produce(const DisplayGeomRecord&);
+
+  static void fillDescriptions(edm::ConfigurationDescriptions&);
+
+private:
+  TGeoManager* createManager(int level);
+
+  TGeoShape* createShape(const std::string& iName, const DDSolid& iSolid);
+  TGeoVolume* createVolume(const std::string& iName, const DDSolid& iSolid, const DDMaterial& iMaterial);
+  TGeoMaterial* createMaterial(const DDMaterial& iMaterial);
+
+  // ---------- member data --------------------------------
+
+  const int m_level;
+  const bool m_verbose;
+  const bool m_fullname;
+  std::string m_TGeoName;
+  std::string m_TGeoTitle;
```

Comment:
@Dr15Jones trivial side remark: that is ok with me, in other PR reviews I have seen suggesting more to use trailing "_" for local variables as an extensive interpretation of our code rules (which in my view say a slightly different thing)